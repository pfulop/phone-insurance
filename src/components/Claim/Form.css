.intro {
  font-size: 24px;
  text-align: center;
  padding: 16px;
  flex: 100%;
}

.ready {
  font-size: 32px;
  color: #28112b;
}

.startClaim {
  background-color: #4e8098;
  text-decoration: none;
  text-transform: none;
  color: white;
  padding: 4px 16px;
  box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2);
  transition: 0.2s;
}

.startClaim:hover {
  box-shadow: 0 0px 0px 0 rgba(0, 0, 0, 0.2);
}

.label {
  line-height: 32px;
}

.input {
  border: 1px solid lightgray;
  border-radius: 4px;
  padding: 6px 12px;
  font-size: 18px;
}

.error {
  padding: 16px;
  line-height: 24px;
  border-radius: 4px;
  color: darkred;
}

.questionArea {
  margin: 16px 0;
}

.nextButton {
  background-color: #4e8098;
  text-decoration: none;
  text-transform: none;
  color: white;
  padding: 4px 16px;
  box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2);
  transition: 0.2s;
  border-radius: 4px;
  border: none;
  font-size: 24px;
  margin-right: 0;
  margin-left: auto;
}

.nextButton:hover {
  box-shadow: 0 0px 0px 0 rgba(0, 0, 0, 0.2);
}

.buttonWrapper {
  margin-top: 8px;
  text-align: right;
}

.title {
  margin-bottom: 16px;
}

.checkboxButton {
  margin: 0;
  margin-top: 2px;
  width: 25px;
  height: 25px;
  appearance: none;
  position: relative;
  border: 2px solid black;
  border-radius: 7px;
  overflow: hidden;
  cursor: pointer;
}
.checkboxButton:after {
  font-size: 29px;
  position: absolute;
  transform: scale(0);
  background-color: transparent;
  border-radius: 50px;
  transition: all 175ms ease-in;
  margin: -7px 0px 0 0px;
}
.checkboxButton:checked:after {
  content: 'âœ“';
  color: black;
  background-color: white;
  border-radius: 0;
  transform: scale(1.3);
}
.checkboxButton:focus {
  outline: 0;
}
