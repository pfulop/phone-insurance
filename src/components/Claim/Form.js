import React, { Component } from 'react';
import css from './Form.css';
import gCss from '../../index.css';
import data from './data.json';

export default class Form extends Component {
  constructor() {
    super();
    this.state = {
      page: data.questions[0].id,
    };
    this.nextPage = this.nextPage.bind(this);
    this.previousPage = this.previousPage.bind(this);
  }

  nextPage() {
    this.setState({ page: this.state.page + 1 });
  }

  previousPage() {
    this.setState({ page: this.state.page - 1 });
  }

  render() {
    const { page } = this.state;

    return (
      <div className={gCss.card}>
        <h1>Claim</h1>

        {/* {page === 1 && <WizardFormFirstPage onSubmit={this.nextPage} />}
        {page === 2 && (
          <WizardFormSecondPage
            previousPage={this.previousPage}
            onSubmit={this.nextPage}
          />
        )} */}
      </div>
    );
  }
}
