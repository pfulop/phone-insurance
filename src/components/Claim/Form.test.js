import React from 'react';
import Form from './Form';
import { shallow } from 'enzyme';
import { createMockStore } from 'redux-test-utils';
import validate from './validations';

const shallowWithStore = (component, store) => {
  const context = {
    store,
  };
  return shallow(component, { context });
};

describe('Form', () => {
  it('should be defined', () => {
    expect(Form).toBeDefined();
  });
  const testState = {
    form: {},
  };
  const store = createMockStore(testState);
  it('should render correctly with store', () => {
    const tree = shallowWithStore(<Form />, store);
    expect(tree).toMatchSnapshot();
  });

  it('should validate string', () => {
    const vals = {
      a: 'a',
    };
    const props = {
      name: 'a',
      type: 'string',
      label: 'a',
    };
    expect(validate(vals, props)).toMatchObject({});
  });

  it('should validate email', () => {
    const vals = {
      a: 'aaa@aaa.aa',
    };
    const props = {
      name: 'a',
      type: 'string',
      label: 'email',
    };
    expect(validate(vals, props)).toMatchObject({});
  });

  it('should validate boolean', () => {
    const vals = {
      a: true,
    };
    const props = {
      name: 'a',
      type: 'boolean',
      label: 'boolean',
    };
    expect(validate(vals, props)).toMatchObject({});
  });

  it('should validate number', () => {
    const vals = {
      a: '123.22',
    };
    const props = {
      name: 'a',
      type: 'number',
      label: 'number',
    };
    expect(validate(vals, props)).toMatchObject({});
  });
});
