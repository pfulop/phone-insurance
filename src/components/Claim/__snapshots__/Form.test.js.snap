// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form should render correctly with store 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Connect(Form) />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "addAnswer": [Function],
      "reset": [Function],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "addAnswer": [Function],
        "reset": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "context": Object {
      "store": Object {
        "dispatch": [Function],
        "getAction": [Function],
        "getActions": [Function],
        "getState": [Function],
        "isActionDispatched": [Function],
        "isActionTypeDispatched": [Function],
        "subscribe": [Function],
      },
    },
  },
}
`;
